src/lib.rs

```diff
@@ -184,40 +184,43 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		/// create a supersig.
 		///
-		/// `create_supersig` will create a supersig with specified parameters, and transfer an
-		/// existencial deposit from the creator to the generated supersig account, to bring the
-		/// account to life.
+		/// `create_supersig` will create a supersig with specified parameters, and transfer
+		/// currencies from the creator to the generated supersig:
+		///     - the existencial deposit (minimum amount to make an account alive)
+		///     - the price corresponding to the size (in bytes) of the members times the
+		///       PricePerBytes
 		///
 		/// The dispatch origin for this call must be `Signed`.
 		///
 		/// # <weight>
 		///
 		/// Related functions:
 		/// - `Currency::transfer` will be called once to deposit an existencial amount on supersig
+		/// - `frame_system::inc_consumers` will be called once to protect the supersig from
+		///   deletion
+		#[transactional]
 		#[pallet::weight(T::WeightInfo::create_supersig())]
 		pub fn create_supersig(origin: OriginFor<T>, members: Vec<T::AccountId>) -> DispatchResult {
 			let who = ensure_signed(origin)?;
-			let members_len = members.len();
-			let supersig = Supersig::new(members).ok_or(Error::<T>::InvalidSupersig)?;
+			let supersig = Supersig::new(members.clone()).ok_or(Error::<T>::InvalidSupersig)?;
 			let index = Self::nonce_supersig();
 			let supersig_id: T::AccountId = T::PalletId::get().into_sub_account(index);
 
-			let minimum_balance = T::Currency::minimum_balance();
-			T::Currency::transfer(
-				&who,
-				&supersig_id,
-				minimum_balance,
-				ExistenceRequirement::KeepAlive,
-			)?;
+			let price = <BalanceOf<T>>::from(size_of::<T::AccountId>() as u32)
+				.saturating_mul((members.len() as u32).into())
+				.saturating_mul(T::PricePerBytes::get());
+			let deposit = price.saturating_add(T::Currency::minimum_balance());
```

Comment:
apparently, its two distinct account under one AccountNonce. so if free_balance become less than existential deposit, i IS deleted, but if there is reseved funds, the AccountNonce is not deleted so we can still make the free balance alive again

But what if ``PricePerByte * [size of members in byte] < ExistencialDeposit`` ? (in case for exemple there is only one user and a verry low pricePerByte.

I think we should still transfer the existencialDeposit, it could protect us from this kind of errors, and it is a very low amount, so it shouldn't hurt the wallet